cpp/gpu/source/python/api.cpp

```diff
@@ -0,0 +1,316 @@
+                            if (heap.flags & vk::MemoryHeapFlagBits::eMultiInstance)
+                                flags.push_back("MULTI_INSTANCE");
+                            if (heap.flags & vk::MemoryHeapFlagBits::eMultiInstanceKHR)
+                                flags.push_back("MULTI_INSTANCE_KHR");
+                            return flags;
+                        }
+                    )
+                    .doc() = "Wrapper around vk::MemoryHeap object.";
+
+                /* Properties of physical device memory retrieved from Vulkan API. */
+                py::class_<vk::MemoryType>(m, "MemoryType")
+                    .def_property_readonly(
+                        "heap_index",
+                        [](const vk::MemoryType& mem_type) {
+                            return mem_type.heapIndex;
+                        }
+                    )
+                    .def_property_readonly(
+                        "flags",
+                        [](const vk::MemoryType& mem_type) {
+                            std::vector<std::string> flags = {};
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eDeviceLocal)
+                                flags.push_back("DEVICE_LOCAL");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eHostVisible)
+                                flags.push_back("HOST_VISIBLE");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eHostCoherent)
+                                flags.push_back("HOST_COHERENT");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eHostCached)
+                                flags.push_back("HOST_CACHED");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eLazilyAllocated)
+                                flags.push_back("LAZILY_ALLOCATED");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eProtected)
+                                flags.push_back("PROTECTED");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eDeviceCoherentAMD)
+                                flags.push_back("DEVICE_COHERENT_AMD");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eDeviceUncachedAMD)
+                                flags.push_back("DEVICE_UNCACHED_AMD");
+                            if (mem_type.propertyFlags &
+                                vk::MemoryPropertyFlagBits::eRdmaCapableNV)
+                                flags.push_back("RDMA_CAPABLE_NV");
+                            return flags;
```

Comment:
Wouldnt it be cleaner if you were checking `if(mem_type.propertyFlags)` only once at the begining instead of repeating it  9 times?